apps/hash-api/src/graph/system-account.ts

```diff
@@ -0,0 +1,41 @@
+import { Logger } from "@local/hash-backend-utils/logger";
+import { systemTypes } from "@local/hash-isomorphic-utils/ontology-types";
+import { AccountId } from "@local/hash-subgraph";
+
+import { publicUserAccountId } from "../graphql/context";
+import { createAccount } from "./knowledge/system-types/account.fields";
+import { getEntityTypeById } from "./ontology/primitive/entity-type";
+import { ImpureGraphContext } from "./util";
+
+// eslint-disable-next-line import/no-mutable-exports
+export let systemAccountId: AccountId;
+
+const authentication = { actorId: publicUserAccountId };
+
+/**
+ * Ensure the `systemAccountId` exists by fetching it or creating it. Note this
+ * method is designed to be run before the system types are initialized.
+ */
+export const ensureSystemAccountExists = async (params: {
+  logger: Logger;
+  context: ImpureGraphContext;
+}) => {
+  const { logger, context } = params;
+
+  try {
+    // The system account is the creator of the system types
+    const orgEntityType = await getEntityTypeById(context, authentication, {
+      entityTypeId: systemTypes.entityType.org.entityTypeId,
+    });
+
+    systemAccountId =
+      orgEntityType.metadata.custom.provenance.recordCreatedById;
+
+    logger.info(`Using existing system account id: ${systemAccountId}`);
+  } catch {
+    // We don't have any system types yet, so we need to create the system account, which will create them
+
+    systemAccountId = await createAccount(context, authentication, {});
+    logger.info(`Created system user account id: ${systemAccountId}`);
+  }
```

Comment:
I think this is only relevant for development, but if an error occurs whilst creating a system type when running this function again a second a system account ID might be created if the `Org` type did not get created on first try. Then we might have some system types with one system account ID, and some system types with other account IDs. It'd be nice if we had some other way of checking whether the system account ID has been created, but I can't think of one unless we hardcode a UUID or something like that.

Reply:
I agree this is a potential problem but I agree it will only happen in dev (no PR will be mergeable if it fails to seed the system types) and it seems that it should be infrequent enough to not worry about. If it happens a couple of times we can rethink.